poh/src/leader_bank_notifier.rs

```diff
@@ -43,6 +61,8 @@ impl LeaderBankNotifier {
         let mut state = self.state.lock().unwrap();
         assert_eq!(state.status, Status::StandBy);
 
+        self.current_bank_id
+            .store(bank.bank_id(), Ordering::Release);
```

Comment:
I think that this can be ::Relaxed? Afaict there is no dependent state we
write/read based on this so we just need atomicity

Reply:
I want to make sure that `current_bank_id` and `state` changes are always visible together so that a consume worker doesn't see that `current_bank_id` changed but it's unable to read the latest `state` change yet. I just moved the release store to after the mutex release to make sure that this condition would hold. Let me know if that makes more sense. I also changed the load to relaxed to make it faster and just act as an atomic

Reply:
Actually I just went with your suggestion and changed it all to relaxed. It's good enough to have an indication that the bank has been updated and we don't need to be perfect about memory access ordering here. It's better to have this check be fast in order to not slow down transaction processing.